### PR TITLE
Incorporated backend functions for randompost and user_post

### PR DESCRIPTION
**Pull request by wilsonwtcheng**

### CODE REVIEW

File: app/assets/javascripts/angular/controllers/posts/userpost.js.erb [hunk 1]
```diff
@@ -2,28 +2,28 @@ app.controller('UserPostCtrl',['$scope', '$http', '$location', '$routeParams', f
...
-  $http.get(url + $routeParams.id + 'posts' ).success(function(response){
+  $http.get(url + $routeParams.user_id + 'posts').success(function(response){
     $scope.post = response;
   });
 
 }]);
 
 
-app.controller('PostCtrl',['$scope', '$http', '$location', function($scope, $http, $location) {
+// app.controller('PostCtrl',['$scope', '$http', '$location', function($scope, $http, $location) {
```
Comment (h8rry) [inline app/assets/javascripts/angular/controllers/posts/userpost.js.erb]: Commented out codes shouldn't be here.

**Comment by h8rry** [comment]
Please gitignore `public/system/`

File: app/assets/templates/posts/userpost.html [hunk 2]
```diff
@@ -1,10 +1,27 @@
 <div class="container">
+  
   <div class="row">
     <h1>Show All Posts by One User</h1>
-
     <h4>Title: {{ post.title }}</h4>
     <h4>Content: {{ post.content }}</h4>
-    <h4>Category: {{ post.category ? post.category : 'null' }}</h4>
```
Comment (f3r) [inline app/assets/templates/posts/userpost.html:7]: Idea: If you intended to only show the category if it has one, perhaps you could use [ng-show](https://docs.angularjs.org/api/ng/directive/ngShow)?

Comment (h8rry) [inline app/assets/templates/posts/userpost.html:7]: Agree

**Comment by h8rry** [comment]
Looks good

**Pull request merged**